src/main/java/org/openhab/tools/analysis/checkstyle/RequireBundleCheck.java

```diff
@@ -44,12 +49,31 @@ protected void processFiltered(File file, List<String> lines) {
             Manifest manifest = new Manifest(new FileInputStream(file));
             Attributes attributes = manifest.getMainAttributes();
 
-            String requireBundleHeaderName = "Require-Bundle";
-            String requireBundleHeaderValue = attributes.getValue(requireBundleHeaderName);
-            if (requireBundleHeaderValue != null) {
+            String fragmentHost = attributes.getValue(FRAGMENT_HOST_HEADER_NAME);
+            String bundleSymbolicName = attributes.getValue(BUNDLE_SYMBOLIC_NAME_HEADER_NAME);
+
+            boolean testBundle = false;
+            if (StringUtils.isNoneBlank(fragmentHost) && StringUtils.isNotBlank(bundleSymbolicName)) {
+                testBundle = bundleSymbolicName.startsWith(fragmentHost)
+                        && bundleSymbolicName.substring(fragmentHost.length()).startsWith(".test");
+            }
+
+            String requireBundleHeaderValue = attributes.getValue(REQUIRE_BUNDLE_HEADER_NAME);
+            if (requireBundleHeaderValue != null && !testBundle) {
                 log(findLineNumber(lines, requireBundleHeaderValue, 0),
                         "The MANIFEST.MF file must not contain any Require-Bundle entries. "
                                 + "Instead, Import-Package must be used.");
+            } else if (requireBundleHeaderValue != null && testBundle) {
+                String[] bundleNames = requireBundleHeaderValue.split(",");
+                for (int i = 0; i < bundleNames.length; i++) {
+                    String bundleName = bundleNames[i];
+                    if (!bundleName.equals("org.junit") && !bundleName.equals("org.hamcrest")
+                            && !bundleName.equals("org.mockito")) {
```

Comment:
It will be great if we add a configuration property that includes the bundles that are excluded. Requirements change over time and can vary between different repositories. Then these bundles will be added to the ruleset.